### PR TITLE
Move MPS loss multiplier out of OD C++ implementation

### PR DESCRIPTION
**Pull request by shreyajain17**

### CODE REVIEW

File: src/ml/neural_net/mps_graph_cnnmodule.mm [hunk 1]
```diff
@@ -115,6 +115,7 @@ - (instancetype)initWithCommandBuffer:(id<MTLCommandBuffer>)commandBuffer {
 
   @autoreleasepool {
 
+  constexpr float MPS_LOSS_MULTIPLIER = 8;
```
Comment (nickjong) [inline src/ml/neural_net/mps_graph_cnnmodule.mm]: Nit: `8.f`

File: src/ml/neural_net/mps_graph_cnnmodule.mm [hunk 2]
```diff
@@ -163,7 +164,7 @@ - (instancetype)initWithCommandBuffer:(id<MTLCommandBuffer>)commandBuffer {
       std::vector<float> loss(loss_size);
       auto loss_it = loss.begin();
       for (MPSCNNLossLabels *lossState in batch.lossState) {
-        *loss_it = sumImage([lossState lossImage]);
+        *loss_it = sumImage([lossState lossImage]) / MPS_LOSS_MULTIPLIER;
```
Comment (nickjong) [inline src/ml/neural_net/mps_graph_cnnmodule.mm]: Could you add an explanatory comment either here or where the constant is defined?

File: src/ml/neural_net/mps_compute_context.mm [hunk 3]
```diff
@@ -88,12 +88,45 @@
 std::unique_ptr<model_backend> mps_compute_context::create_object_detector(
     int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
     const float_array_map& config, const float_array_map& weights) {
-
+  float_array_map updated_config;
+  constexpr float MPS_LOSS_MULTIPLIER = 8;
```
Comment (nickjong) [inline src/ml/neural_net/mps_compute_context.mm]: Let's add a comment here. Also, we should be able to define the constant in only one place, probably in `mps_graph_cnn_module.h`

Can probably just define `static constexpr float MPS_OD_LOSS_MULTIPLIER = 8.f;` somewhere

File: src/ml/neural_net/mps_compute_context.mm [hunk 4]
```diff
@@ -88,12 +88,45 @@
...
+  if (config.count("learning_rate") > 0) {
+    updated_config["learning_rate"] =
+      shared_float_array::wrap(*config.at("learning_rate").data() / MPS_LOSS_MULTIPLIER);
+  }
```
Comment (nickjong) [inline src/ml/neural_net/mps_compute_context.mm]: Could we define a quicker helper function or lambda here?

Also, prefer this pattern:
```
auto config_iter = config.find("learning_rate");
if (config_iter != config.end()) {
  float old_value = config_iter->second.data()[0];
  float new_value = old_value / MPS_LOSS_MULTIPLIER;
  config_iter->second = shared_float_array::wrap(new_value);
}

File: src/ml/neural_net/mps_compute_context.mm [hunk 5]
```diff
@@ -20,6 +20,25 @@
 
 namespace {
 
+float_array_map multiply_mps_od_loss_multiplier(float_array_map config,
+                                                std::vector<std::string> update_keys) {
```
Comment (nickjong) [inline src/ml/neural_net/mps_compute_context.mm]: Nit: pass `update_keys` by `const` reference

File: src/ml/neural_net/mps_compute_context.mm [hunk 6]
```diff
@@ -20,6 +20,25 @@
...
+  for (std::vector<std::string>::const_iterator i = update_keys.begin(); i != update_keys.end();
```
Comment (nickjong) [inline src/ml/neural_net/mps_compute_context.mm]: This would be more readable as:
```
for (const std::string& key : update_keys) {
```
